x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/assembler/ExecutionManager.java

```diff
@@ -65,13 +68,30 @@ public Executable assemble(List<List<Attribute>> listOfKeys,
         for (int i = 0; i < plans.size(); i++) {
             List<Attribute> keys = listOfKeys.get(i);
             List<HitExtractor> keyExtractors = hitExtractors(keys, extractorRegistry);
+            List<String> keyFields = new ArrayList<>(keyExtractors.size());
+
+            // extract top-level fields used as keys to optimize query lookups
+            // this process gets skipped for nested fields
+            for (HitExtractor extractor : keyExtractors) {
+                if (extractor instanceof AbstractFieldHitExtractor) {
+                    AbstractFieldHitExtractor hitExtractor = (AbstractFieldHitExtractor) extractor;
+                    // no nested fields
+                    if (hitExtractor.hitName() == null) {
+                        keyFields.add(hitExtractor.fieldName());
+                    }
```

Comment:
`else {` should be on the same line as the closing curly bracket from the `if` statement, no?